src/Stratis.CirrusMinerD/Program.cs

```diff
@@ -0,0 +1,126 @@
+using Stratis.Bitcoin.Configuration;
+using Stratis.Bitcoin.Features.Api;
+using Stratis.Bitcoin.Features.BlockStore;
+using Stratis.Bitcoin.Features.Consensus;
+using Stratis.Bitcoin.Features.MemoryPool;
+using Stratis.Bitcoin.Features.Miner;
+using Stratis.Bitcoin.Features.Notifications;
+using Stratis.Bitcoin.Features.RPC;
+using Stratis.Bitcoin.Features.SmartContracts;
+using Stratis.Bitcoin.Features.SmartContracts.PoA;
+using Stratis.Bitcoin.Features.SmartContracts.Wallet;
+using Stratis.Bitcoin.Features.Wallet;
+using Stratis.Bitcoin.Networks;
+using Stratis.Bitcoin.Utilities;
+using Stratis.Features.FederatedPeg;
+using Stratis.Sidechains.Networks;
+
+namespace Stratis.CirrusMinerD
+{
+    class Program
+    {
+        private const string MainchainArgument = "-mainchain";
+        private const string SidechainArgument = "-sidechain";
+
+        private static readonly Dictionary<NetworkType, Func<Network>> MainChainNetworks = new Dictionary<NetworkType, Func<Network>>
+        {
+            { NetworkType.Mainnet, Networks.Stratis.Mainnet },
+            { NetworkType.Testnet, Networks.Stratis.Testnet },
+            { NetworkType.Regtest, Networks.Stratis.Regtest }
+        };
+
+        public static void Main(string[] args)
+        {
+            MainAsync(args).Wait();
+        }
+
+        public static async Task MainAsync(string[] args)
+        {
+            try
+            {
+                bool isMainchainNode = args.FirstOrDefault(a => a.ToLower() == MainchainArgument) != null;
+                bool isSidechainNode = args.FirstOrDefault(a => a.ToLower() == SidechainArgument) != null;
+
+                if (isSidechainNode == isMainchainNode)
+                {
+                    throw new ArgumentException($"Gateway node needs to be started specifying either a {SidechainArgument} or a {MainchainArgument} argument");
+                }
+
+                IFullNode node = isMainchainNode ? GetStratisNode(args) : GetCirrusMiningNode(args);
+
+                if (node != null)
+                    await node.RunAsync();
+            }
+            catch (Exception ex)
+            {
+                Console.WriteLine("There was a problem initializing the node. Details: '{0}'", ex.Message);
+            }
+        }
+
+        private static IFullNode GetCirrusMiningNode(string[] args)
+        {
+            var nodeSettings = new NodeSettings(networksSelector: CirrusNetwork.NetworksSelector, protocolVersion: ProtocolVersion.ALT_PROTOCOL_VERSION, args: args)
+            {
+                MinProtocolVersion = ProtocolVersion.ALT_PROTOCOL_VERSION
+            };
+
+            IFullNode node = new FullNodeBuilder()
+                .UseNodeSettings(nodeSettings)
+                .UseBlockStore()
+                .SetCounterChainNetwork(MainChainNetworks[nodeSettings.Network.NetworkType]())
+                .UseSmartContractPoAConsensus()
+                .UseSmartContractPoAMining()
+                .CheckForCollateral()
```

Comment:
this adds collaterack checker but it seems it's not initialized. The only place where we call initialization is from fed peg feature which is not added here.

Btw have you tested this new daemon in a real setup?

Reply:
Have amended it to call the collateral checker in CollateralFeature.

Reply:
Tested it all locally, no issues. The collateral members on CirrusMain are all Money(0) though so it wasn't searching for any collateral I guess but it was passing through all the code inside `CollateralChecker`